docs/api/csf/csf-factories.mdx

```diff
@@ -0,0 +1,305 @@
+
+```diff title="Button.stories.ts"
+// ...rest of file
+
++ export const Primary = meta.story({
+- export const Primary: Story = {
+    args: { primary: true },
++ });
+- };
+
++ export const PrimaryDisabled = meta.story({
+- export const PrimaryDisabled: Story = {
+    args: {
++     ...Primary.composed.args,
+-     ...Primary.args,
+      disabled: true,
+    }
++ });
+- };
+```
+
+<Callout variant="info">
+  The property name "composed" was chosen because the values within are composed from the story, its component meta, and the preview configuration.
+
+  If you wish instead to access only the direct input to the story, you can use `Story.input` instead of `Story.composed`.
+</Callout>
+
+### 5. Reusing stories in test files
+
+[Storybook's Test addon](../../writing-tests/test-addon.mdx) allows you to test your components directly inside Storybook. All of the stories are automatically turned into Vitest tests, so the integration is seamless in your testing suite.
+
+If you cannot use Storybook Test, you can still reuse the stories in your test files using [portable stories](../portable-stories/portable-stories-vitest.mdx). In prior story formats, you had to compose the stories before rendering them in your test files. With CSF Factories, you can now reuse the stories directly.
+
+```diff title="Button.test.ts"
+import { test, expect } from 'vitest';
+import { screen } from '@testing-library/react';
+- import { composeStories } from '@storybook/react';
+ 
+// Import all stories from the stories file
+import * as stories from './Button.stories';
+ 
++ const { Primary } = stories;
+- const { Primary } = composeStories(stories);
+ 
+test('renders primary button with default args', async () => {
+  // The run function will mount the component and run all of Storybook's lifecycle hooks
+  await Primary.run();
+  const buttonElement = screen.getByText('Text coming from args in stories file!');
+  expect(buttonElement).not.toBeNull();
+});
+```
+
+The `Story` object also provides a `Component` property in case you want to render it using another method of your choosing, for instance, with [Testing Library](https://testing-library.com/). You also have access to its composed properties (args, parameters, etc.) via the `composed` property.
+
+Here's an example of how you can reuse a story in a test file by rendering its component:
+
+{/* prettier-ignore-start */}
+<CodeSnippets path="portable-stories-csf-factory-render.md" />
+{/* prettier-ignore-end */}
+
+## Frequently asked questions (FAQ)
+
+### Will I have to migrate all of my stories to this new format?
+
+Storybook will continue to support CSF 1, [CSF 2](../../../release-6-5/docs/api/stories/csf.mdx), and [CSF 3](./index.mdx) for the foreseeable future. None of these prior formats are deprecated.
```

Comment:
```suggestion
Storybook will continue to support CSF 1, [CSF 2](../../../release-6-5/docs/api/stories/csf.mdx), and [CSF 3](./index.mdx) for the foreseeable future. None of these prior formats is deprecated.
```

Reply:
This is incorrect. The deprecated state applies to the formats, which is plural, so "are" is correct.